File: src/main/java/shopping/dto/response/ProductResponse.java
package shopping.dto.response;

import shopping.domain.entity.Product;

public class ProductResponse {

    private final Long id;
    private final String name;
    private final String image;
    private final long price;

    public ProductResponse(final Long id, final String name, final String image, final long price) {
        this.id = id;
        this.name = name;
        this.image = image;
        this.price = price;
    }

    public static ProductResponse of(final Product product) {
        return new ProductResponse(
                product.getId(),
                product.getName().getValue(),
                product.getImage().getValue(),
                product.getPrice().getValue()
        );
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getImage() {
        return image;
    }

    public long getPrice() {
        return price;
    }
}

File: src/main/kotlin/shopping/web/product/response/ProductResponse.kt
package shopping.web.product.response

import shopping.core.product.Product

data class ProductResponse(val name: String, val price: Long, val imageUrl: String, val id: Long) {
    companion object {
        fun fromDomain(product: Product): ProductResponse {
            return ProductResponse(
                name = product.name,
                price = product.price,
                imageUrl = product.imageUrl,
                id = product.id,
            )
        }
    }
}
